src/main/java/mate.academy/lib/Injector.java

```
@@ -1,13 +1,73 @@
 package mate.academy.lib;
 
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Field;
+import java.util.HashMap;
+import java.util.Map;
+import mate.academy.service.FileReaderService;
+import mate.academy.service.ProductParser;
+import mate.academy.service.ProductService;
+import mate.academy.service.impl.FileReaderServiceImpl;
+import mate.academy.service.impl.ProductParserImpl;
+import mate.academy.service.impl.ProductServiceImpl;
+
 public class Injector {
     private static final Injector injector = new Injector();
+    private final Map<Class<?>, Object> instances = new HashMap<>();
 
     public static Injector getInjector() {
         return injector;
     }
 
     public Object getInstance(Class<?> interfaceClazz) {
-        return null;
+        Object clazzImplementationInstance = null;
+        Object fieldInstance = null;
+        Class<?> clazz = findImplementation(interfaceClazz);
```

This violates checklist item #4: "Don't forget about annotations checks in your Injector class". Right after finding the implementation class, you must check if it's annotated with `@Component`. If the annotation is missing, the requirements state that you should throw an exception. This check is a crucial part of the injector's logic and is currently missing.